### PR TITLE
FIX - customize theme

### LINKED ISSUE #712: [BUG]  '@link-color' does not effect the mouse-over color in details section
**Describe the bug**

config.json property '@link-color' changes navigation section mouse-over color of links, but not the mouse-over colour in the details.

**To Reproduce**
Steps to reproduce the behavior; '@link-color' in the config.json file.

1. change the property 

**Expected behavior**

The mouse-over colour in the details section changes to match the color set in the config.json file

**Screenshots**

image

**Additional context**

### LINKED ISSUE #713: [BUG] '@processing' color does not affect the page loading progress or circular 'I'm thinking' graphic
**Describe the bug**

**To Reproduce**

Change the property  '@processing' in the config.json file.

1. go to Primate URL (see large 'thinking' graphic is still blue)
2. Log in as user or admin.  (see small 'thinking' graphic and page loading bar at the top of the page are still blue)

**Expected behaviour**

- 'thinking' graphic and page loading bar should match the '@processing' bar.

**Screenshots**

image

**Additional context**

### LINKED ISSUE #714: [BUG] (UI) (borderline feature request) - can't change the navigation menu color for default view or project view
**Describe the bug**

**To Reproduce**

none - can't do it...

**Expected behavior**

To match all other branding, a cloud operator will likely want to change the navigation background colors.

**Screenshots**

image  

image

**Additional context**

### PR DESCRIPTION
**Pull request by utchoang**
Fixes #712
Fixes #713 
Fixes #714

- [x] Customize link hover color
- [x] Customize loading color
- [x] Customize navigation menu color

@rhtyd cc @svenvogel

### CODE REVIEW

**Comment by utchoang** [comment]
- Customize link hover color

image

- Customize loading color

image

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by utchoang** [comment]
@rhtyd Do we need to add a light / dark mode to the red border area (image)? Because this will make CSS editing pretty much.
image

**Comment by utchoang** [comment]
@rhtyd Does the dark mode of the switch to the project affect the custom theme with config.json? For example background color?
=

**Comment by utchoang** [comment]
Customize navigation background color and text color
image
image

**Comment by utchoang** [comment]
Customize navigation color
image

**Comment by utchoang** [comment]
@blueorangutan package

**Comment by utchoang** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
LGTM, need bit of testing 
@davidjumani @andrijapanicsb can you review/test? Thanks

**Comment by rhtyd** [comment]
cc @DaanHoogland

**Review by rhtyd — APPROVED** [review]

**Comment by rhtyd** [comment]
@utchoang code LGTM, can you update the https://github.com/apache/cloudstack-primate/blob/master/docs/customize.md in sync with your FR and the doc https://github.com/apache/cloudstack-documentation/blob/master/source/installguide/primate.rst (you'll need to send a separate PR to the apache/cloudstack-documentation repo). Thanks.

I'll need second pair of testing from @DaanHoogland or @davidjumani thanks

**Review by DaanHoogland — COMMENTED** [review]
I cannot add sensible comments about the code but will give it some test attention

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]
tested and looks good.

**Comment by DaanHoogland** [comment]
@rhtyd do we (a.k.a @utchoang) update the customize.md before merging?

**Comment by DaanHoogland** [comment]
@utchoang I saw that only the navigation colour can be set. I would expect the main pane to be matching. Does that make sense? Or should it also be a separate colour?

**Comment by utchoang** [comment]
@DaanHoogland Regarding the adjustment of the main page, it takes quite a lot of changes (CSS, style). Can I do it if it's absolutely necessary? @rhtyd thought?

**Comment by DaanHoogland** [comment]
@utchoang It was just an observation with regards to the issue #714 I do consider this a feature (not broderline-) so I do not regard this part of 1.0 ga! I think we should have a full theming/cusitmization issue for 1.1. I think we should merge this and call it a release. I assure you, I'll keep coming up with issues if we don't.

**Comment by DaanHoogland** [comment]
are you updating `customize.md` in this PR, @utchoang ?

**Comment by utchoang** [comment]
@DaanHoogland Yes. I've updated.

**Comment by rhtyd** [comment]
Thanks @utchoang @DaanHoogland lgtm, but I need to do one round of test with a real deb/rpm based package. I'll merge it after that (today/tomorrow).

**Comment by rhtyd** [comment]
@utchoang can you send a doc PR to https://github.com/apache/cloudstack-documentation/blob/master/source/installguide/primate.rst#basic-customization-in-cloudstack-primate

**Comment by rhtyd** [comment]
@blueorangutan package

File: src/views/AutogenView.vue
Comment (rhtyd) [inline src/views/AutogenView.vue:656]: @utchoang why not dispatch logout on 401/unauthorise error?

File: src/views/compute/DeployVM.vue
Comment (rhtyd) [inline src/views/compute/DeployVM.vue:1752]: @utchoang why is this part of the PR change?

File: src/views/iam/DomainView.vue
Comment (rhtyd) [inline src/views/iam/DomainView.vue:184]: @utchoang same as above?

File: src/views/AutogenView.vue
Comment (utchoang) [reply src/views/AutogenView.vue:659]: @rhtyd I don't have changed it. Maybe it's because I haven't merged with the latest master yet.

**Comment by rhtyd** [comment]
Tested, LGTM
Verified left nav bar colour for default view:
Screenshot from 2020-09-23 11-26-11

Project view is still dark-er though:
Screenshot from 2020-09-23 11-29-26

@utchoang would it be possible to define a custom background colour for the navigation bar on left when we're in a project view (i.e. selected a project)?

**Comment by utchoang** [comment]
* For select project
image

**Comment by rhtyd** [comment]
Thanks @utchoang pl update the customize.md and send a doc PR to the documentation repo

**Review by rhtyd — APPROVED** [review]

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by utchoang** [comment]
@rhtyd I've updated customize.md on this PR and a document on https://github.com/apache/cloudstack-documentation/pull/159

**Comment by rhtyd** [comment]
Thanks @utchoang I'll review, test and merge shortly

**Pull request merged**
